src/backer/cardaning.py

```diff
@@ -104,122 +98,130 @@ def flushQueue(self):
             self.timer = Timer(90, self.flushQueue)
             self.timer.start()
 
-    def getaddressBalance(self):
-        try:
-            address = self.api.address(address=self.spending_addr.encode())
-            return int(address.amount[0].quantity)
-        except ApiError as e:
-            return 0
+    def getaddressBalance(self, addr):
+        try:            
+            utxo_list = self.client.query_utxo.execute([ogmios.Address(addr)])       
+
+            if utxo_list and len(utxo_list[0]) > 0:          
+                utxo = utxo_list[0][0]                
+                return int(utxo.value['ada']['lovelace'])
+            
+        except Exception as e:
+            logger.debug(traceback.format_exc())
+            logger.info(f"Error: Cannot get address's balance: {e}")
```

Comment:
`.critical`

Reply:
removed the "print" and use logger.critical instead